Azure/azure-cli
# {Cloud} Improve endpoint discovery from metadata url env variable

**Description**  

When env var `ARM_CLOUD_METADATA_URL` is set, improve performance by caching cloud endpoints loaded from the URL in local file. The file will be refreshed when Azure CLI is updated.

Fix the issues that the default cloud is always set to AzureCloud.

**Testing Guide**  

`export ARM_CLOUD_METADATA_URL=https://management.azure.com/metadata/endpoints?api-version=2019-05-01`

Run any az command, you should see cloud endpoints stored in `~/.azure/cloudEndpoints.json`.

**History Notes**  


[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.  
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.
